### PR TITLE
BACK-928 : Lama logging

### PR DESCRIPTION
**Pull request by jjayet-ledger**
## What is this about?

Adds logs to Lama

## Cute picture of lama
44337b67caa020912aaf83c6dc5fb2f0

### CODE REVIEW

**Comment by estrauser-ledger** [comment]
LGTM

File: coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Queries.scala [hunk 1]
```diff
@@ -5,21 +5,27 @@ import java.util.UUID
...
+
     for {
-      tx      <- fetchTxAndBlock(accountId, hash)
-      inputs  <- fetchInputs(accountId, hash).compile.toList
+      tx <- fetchTxAndBlock(accountId, hash)
+      _ = log.info(s"Transaction $tx")
+      inputs <- fetchInputs(accountId, hash).compile.toList
+      _ = log.info(s"Inputs $inputs")
       outputs <- fetchOutputs(accountId, hash).compile.toList
+      _ = log.info(s"Outputs $outputs")
```
Comment (estrauser-ledger) [inline coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Queries.scala:28]: I'd debug these logs

File: coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala [hunk 2]
```diff
@@ -20,10 +21,11 @@ trait Interpreter extends BitcoinInterpreterServiceFs2Grpc[IO, Metadata] {
   def bytesToUUID(bytes: ByteString): UUID = UUID.nameUUIDFromBytes(bytes.toByteArray)
 }
 
-class DbInterpreter(db: Transactor[IO]) extends Interpreter {
+class DbInterpreter(db: Transactor[IO]) extends Interpreter with IOLogging {
 
   def saveTransactions(request: SaveTransactionsRequest, ctx: Metadata): IO[Empty] = {
 
+    log.info(s"Saving transactions: ${request.transactions}")
```
Comment (estrauser-ledger) [inline coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala]: there may be times where hundreds of transactions come at once, maybe log the size instead ?

File: coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala [hunk 3]
```diff
@@ -57,6 +70,8 @@ class DbInterpreter(db: Transactor[IO]) extends Interpreter {
           .compile
           .toList
 
+      _ = log.info(s"$ops")
```
Comment (estrauser-ledger) [inline coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala]: maybe not log every operations fetched

File: coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala (hunk 2 above)
Comment (jjayet-ledger) [reply coins/bitcoin/interpreter/src/main/scala/co/ledger/lama/bitcoin/interpreter/Interpreter.scala]: :+1:

**Comment by estrauser-ledger** [comment]
LGTM

**Review by nukky94 — APPROVED** [review]

**Pull request merged**